osu.Game.Rulesets.Catch/UI/CatcherArea.cs

```diff
@@ -25,15 +25,13 @@ public class CatcherArea : Container, IKeyBindingHandler<CatchAction>
         public Catcher Catcher
         {
             get => catcher;
-            set
-            {
-                if (catcher != null)
-                    Remove(catcher);
-
-                Add(catcher = value);
-            }
+            set => catcherContainer.Child = catcher = value;
         }
 
+        internal CatcherTrailDisplay CatcherTrails { get; }
```

Comment:
Exposition as `internal` for tests could potentially be avoided by using `ChildrenOfType<>()`.